pkg/utils/utils.go

```diff
@@ -507,3 +507,18 @@ func dig[T any](key string, data map[string]any) (t T, _ error) {
 
 	return t, errKeyNotFound
 }
+
+func CompareStringAndByteSlices(valueString *string, valueByte []byte) bool {
+	stringToByteSlice := []byte(*valueString)
+	if len(stringToByteSlice) != len(valueByte) {
+		return false
+	}
+
+	for sb := range valueByte {
+		if stringToByteSlice[sb] != valueByte[sb] {
+			return false
+		}
+	}
+
```

Comment:
this function would currently break users with `secretBinary` set (as you can see with the unit tests failures). I think an if is needed here:
```suggestion

func CompareStringAndByteSlices(valueString *string, valueByte []byte) bool {
        if valueString == nil {
          return false
         }
	stringToByteSlice := []byte(*valueString)
	if len(stringToByteSlice) != len(valueByte) {
		return false
	}

	for sb := range valueByte {
		if stringToByteSlice[sb] != valueByte[sb] {
			return false
		}
	}
```
You can check if this worked by running the unit tests locally with `make test`

Reply:
Good catch! wonder how my local tests passed. Fix is on the way.